### PR TITLE
f-searchbox@v4.0.0beta.14

### PR DESCRIPTION
**Pull request by kevinrodrigues**
PR introduces Loqate to searchbox. This covers the initial call to Loqate & displaying results.

### Added
- Loqate core functionality, initial Axios call & display of results.
- Tests to cover Loqate functionality.

### Screenshot
Screen Shot 2021-01-07 at 1 47 36 PM

### CODE REVIEW

File: packages/components/molecules/f-searchbox/src/components/Form.vue [hunk 1]
```diff
@@ -143,13 +146,13 @@ export default {
         /**
          * Display API suggestions in component: `form-search-suggestions`:
          *
-         * 1. Service layer should contain `autocomplete`.
+         * 1. Service layer should contain `autocomplete` or `isFullAddressSearchEnabled`.
```
Comment (ashleynolan) [inline packages/components/molecules/f-searchbox/src/components/Form.vue]:
```suggestion
         * 1. Service layer should contain `isAutocompleteEnabled ` or `isFullAddressSearchEnabled`.
```

Comment (kevinrodrigues) [reply packages/components/molecules/f-searchbox/src/components/Form.vue]: This one is actually correct as it maps to an option called `autocomplete`. I have updated the comment so it's a little clearer.

**Review by dandel10n — DISMISSED** [review]

File: packages/components/molecules/f-searchbox/src/components/Form.vue [hunk 2]
```diff
@@ -30,7 +30,8 @@
                 :is-compressed="isCompressed" />
         </div>
 
-        <form-search-suggestions
+        <component
+            :is="setSuggestionType()"
```
Comment (DamianMullins) [inline packages/components/molecules/f-searchbox/src/components/Form.vue]: Do you need the parentheses here?

File: packages/components/molecules/f-searchbox/src/components/formElements/FormFullAddressSearchSuggestions.vue [hunk 3]
```diff
@@ -0,0 +1,199 @@
+<template>
+    <div :class="$style.shell">
+        <button
+            v-for="(item, index) in getAddressItems"
+            :key="index"
```
Comment (DamianMullins) [inline packages/components/molecules/f-searchbox/src/components/formElements/FormFullAddressSearchSuggestions.vue]: Best to give the key a value which is more likely to be unique i.e. ``:key="`${index}_ getAddressItem`"``

File: packages/components/molecules/f-searchbox/src/components/Form.vue (hunk 2 above)
Comment (kevinrodrigues) [reply packages/components/molecules/f-searchbox/src/components/Form.vue]: It was a method but probably should have been a computed prop so updated it now 👍

File: packages/components/molecules/f-searchbox/src/components/formElements/FormFullAddressSearchSuggestions.vue (hunk 3 above)
Comment (kevinrodrigues) [reply packages/components/molecules/f-searchbox/src/components/formElements/FormFullAddressSearchSuggestions.vue]: Yup, fixed.

File: packages/components/molecules/f-searchbox/src/components/Form.vue (hunk 2 above)
Comment (DamianMullins) [reply packages/components/molecules/f-searchbox/src/components/Form.vue]: Ah yeah, that makes sense 👍🏻

File: packages/components/molecules/f-searchbox/src/store/searchbox.module.js [hunk 4]
```diff
@@ -174,6 +185,27 @@ export default {
...
+         * @param suggestions
+         */
+        [SET_PARTIAL_ADDRESS_SUGGESTIONS]: (state, suggestions) => {
+            state.suggestions = suggestions.Items.map(({
+                Description,
+                Text,
+                Id,
+                Type
+            }) => ({
+                Description,
```
Comment (DamianMullins) [inline packages/components/molecules/f-searchbox/src/store/searchbox.module.js:204]: You _could` map these into camel case? Or would that cause issues elsewhere?

**Review by DamianMullins — DISMISSED** [review]

File: packages/components/molecules/f-searchbox/src/store/searchbox.module.js (hunk 4 above)
Comment (kevinrodrigues) [reply packages/components/molecules/f-searchbox/src/store/searchbox.module.js:204]: WIll take a look in the next PR if that's ok. I think i did try that in the old searchbox but it was a while back so can't remember if it worked ok or not :D

**Review by dandel10n — DISMISSED** [review]

**Review by DamianMullins — APPROVED** [review]

**Review by dandel10n — APPROVED** [review]

**Comment by zlatinivanov-je** [comment]
All good to me. Just missed out on adding a review. 👍

**Pull request merged**